tests/unit/test_discovery_client.py

```diff
@@ -68,6 +75,63 @@ def test___discovery_service_unavailable___register_service_registration_failure
     assert ~unregistration_success_flag  # False
 
 
+def test____get_discovery_service_address___start_service_jit___returns_expected_value(
+    mocker: MockerFixture,
+):
+    mock_key_file_content = {"SecurePort": "", "InsecurePort": _TEST_SERVICE_PORT}
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+        return_value=_FAKE_DISCOVERY_SERVICE_KEY_FILE_PATH,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client.json.load",
+        return_value=mock_key_file_content,
+    )
+    mocker.patch("ni_measurementlink_service._internal.discovery_client._open_key_file")
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
+        return_value=_FAKE_DISCOVERY_SERVICE_KEY_FILE_PATH,
+    )
+    mock_exe_file_path = mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_discovery_service_location",
+        return_value=_FAKE_DISCOVERY_SERVICE_KEY_FILE_PATH,
+    )
+    mock_popen = mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client.subprocess.Popen"
+    )
+
+    popen_args = [[mock_exe_file_path.return_value], mock_exe_file_path.return_value.parent]
+    discovery_service_address = _get_discovery_service_address()
+
+    assert mock_popen.call_args_list[0].args[0] == popen_args[0]
+    assert _TEST_SERVICE_PORT in discovery_service_address
+
+
+def test___get_discovery_service_address___key_file_not_exist___throws_timeouterror(
+    mocker: MockerFixture,
+):
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+        return_value=_FAKE_DISCOVERY_SERVICE_KEY_FILE_PATH,
+    )
+    mocker.patch(
+        "ni_measurementlink_service._internal.discovery_client._open_key_file", side_effect=IOError
```

Comment:
You are testing two cases:
- Polling succeeds on the first try
- Polling never succeeds

To test "polling succeeds after N tries", it's reasonable to patch `_open_key_file` to specify `side_effect=` either a list of exceptions & return values (e.g. `side_effect=[IOError, IOError, IOError, KEY_FILE_PATH.open()]`) or a callable.

Reply:
Added a new method to test "polling succeeds after N tries"